zapisy/apps/grade/poll/models.py

```diff
@@ -117,10 +154,19 @@ def serialize_for_signing_protocol(self) -> dict:
 
         :returns: a dictionary consisting of 'id', 'name' and 'type' keys.
         """
-        result = {'id': self.pk, 'name': self.subcategory, 'type': self.category}
+        result = {"id": self.pk, "name": self.subcategory, "type": self.category}
 
         return result
 
+    def to_dict(self, number_of_submissions=0):
+        return {
+            "id": self.pk,
+            "category": self.category,
+            "subcategory": self.subcategory,
+            "number_of_submissions": number_of_submissions,
+            "is_own": self._is_own
+        }
```

Comment:
Zależałoby mi na tym, żeby nie dokładać pól/propercji `is_own`/`_is_own` do modelu `Poll` (przede wszystkim dlatego, że przeciążanie konstruktora modelu Django wydaje mi się – nawet jeśli na zapas – niepotrzebnym ryzykiem nieprzewidzianych konsekwencji; w szczególności nie robimy tego nigdzie indziej w projekcie).

Pomysł na poprawkę (być może jeszcze nie najbardziej elegancką, ale całość stanie się klarowniejsza do dalszych rozważań) byłby taki, by ta metoda wyglądała tak:
```suggestion
    def to_dict(self, number_of_submissions=0, is_own=False):
        return {
            "id": self.pk,
            "category": self.category,
            "subcategory": self.subcategory,
            "number_of_submissions": number_of_submissions,
            "is_own": is_own
        }
```
i potem była wywoływana z `is_own=(request.user.employee in [poll.owner, poll.teacher, poll.gcowner])` lub podobnie. Oczywiście to może znowu wymagać pewnego przebudowania funkcji `group` itp.